### ISSUE TITLE
f-string comma parsed differently than python3.9

### ISSUE THREAD

**Issue by jepler** [micropython/micropython#9635]
A tuple without enclosing parens can be printed by f-string in Python 3.9:
```
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> f"{1,=}"
'1,=(1,)'
```

In MicroPython, it is a SyntaxError:
```
MicroPython v1.19.1-544-g89b320737-dirty on 2022-10-15; linux [GCC 10.2.1] version
Use Ctrl-D to exit, Ctrl-E for paste mode
>>> f"{1,=}"
Traceback (most recent call last):
  File "<stdin>", line 1
SyntaxError: invalid syntax
```

**Comment by jimmo** [comment]
Thanks @jepler 

This confused me initially because we don't do anything special with commas, but of course what's happening is the f-string transform does:

`f"{1,=}"` --> `"1,={}".format(1,,)`

So it would be worse if we didn't lazily include the trailing comma in the format args, because it would be incorrect rather than syntax error.

The fix is quite simple -- enclose each arg to `format()` in parens (two line change, see https://github.com/micropython/micropython/commit/8f52b545495f541e694af9950a7c73406e85201b )   (+16 bytes FWIW)

Did this come up in real code, is it worth fixing?

**Comment by jepler** [comment]
What I _really_ wanted was repr-formatted `f"{arg=!r}"` which is a SyntaxError in MP/CP. Since tuple items are repr'd I thought adding the comma was a quick way to get what I wanted, but it didn't work out either.

So I'm not super interested in adding it, I'd rather bank 16 bytes against adding !r to f-strings.

Is it worth documenting as a difference, though?

**Comment by jimmo** [comment]
> An adjacent mis-behavior that's more likely to bite someone:

@jepler Just thought of this too last night... :) Yeah I think we should consider fixing this then. I will send a PR with that commit above.

**Comment by dpgeorge** [comment]
Fixed by fb8792c095802a9fcc8b38c1d8cbc3f641918433
